org/mozilla/jss/ssl/javax/JSSEngine.java

```diff
@@ -603,6 +616,12 @@ public void setKeyMaterials(PK11Cert our_cert, PK11PrivKey our_key) throws Illeg
             throw new IllegalArgumentException("JSSEngine.setKeyMaterials(): Either both cert and key must be null or both must be not-null");
         }
 
+        if (ssl_fd != null) {
```

Comment:
I don't think this should be quite correct. We might rehandshake and need client authentication.